src/foam/u2/view/ChoiceWithOtherView.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```

Comment:
In the future I think we can just inline this into the ChoiceView as an optional feature.